src/main/java/com/github/manolo8/darkbot/utils/DiscordWebhook.java

```diff
@@ -0,0 +1,339 @@
+
+                jsonEmbed.addProperty("title", embed.getTitle());
+                jsonEmbed.addProperty("description", embed.getDescription());
+                jsonEmbed.addProperty("url", embed.getUrl());
+
+                if (embed.getColor() != null) {
+                    Color color = embed.getColor();
+                    int rgb = color.getRed();
+                    rgb = (rgb << 8) + color.getGreen();
+                    rgb = (rgb << 8) + color.getBlue();
+
+                    jsonEmbed.addProperty("color", rgb);
+                }
+
+                EmbedObject.Footer footer = embed.getFooter();
+                EmbedObject.Image image = embed.getImage();
+                EmbedObject.Thumbnail thumbnail = embed.getThumbnail();
+                EmbedObject.Author author = embed.getAuthor();
+                List<EmbedObject.Field> fields = embed.getFields();
+
+                if (footer != null) {
+                    JsonObject jsonFooter = new JsonObject();
+
+                    jsonFooter.addProperty("text", footer.getText());
+                    jsonFooter.addProperty("icon_url", footer.getIconUrl());
+                    jsonEmbed.add("footer", jsonFooter);
+                }
+
+                if (image != null) {
+                    JsonObject jsonImage = new JsonObject();
+
+                    jsonImage.addProperty("url", image.getUrl());
+                    jsonEmbed.add("image", jsonImage);
+                }
+
+                if (thumbnail != null) {
+                    JsonObject jsonThumbnail = new JsonObject();
+
+                    jsonThumbnail.addProperty("url", thumbnail.getUrl());
+                    jsonEmbed.add("thumbnail", jsonThumbnail);
+                }
+
+                if (author != null) {
+                    JsonObject jsonAuthor = new JsonObject();
+
+                    jsonAuthor.addProperty("name", author.getName());
+                    jsonAuthor.addProperty("url", author.getUrl());
+                    jsonAuthor.addProperty("icon_url", author.getIconUrl());
+                    jsonEmbed.add("author", jsonAuthor);
+                }
+
+                List<JsonObject> jsonFields = new ArrayList<>();
+                for (EmbedObject.Field field : fields) {
+                    JsonObject jsonField = new JsonObject();
+
+                    jsonField.addProperty("name", field.getName());
+                    jsonField.addProperty("value", field.getValue());
+                    jsonField.addProperty("inline", field.isInline());
+
+                    jsonFields.add(jsonField);
+                }
+//                jsonEmbed.add("fields", jsonFields.toArray());
+                jsonFields.forEach(jsonField -> jsonEmbed.add("fields", jsonField));
+                embedObjects.add(jsonEmbed);
+            }
+//            json.add("embeds", embedObjects.toArray());
+            embedObjects.forEach(embedObject -> json.add("embeds", embedObject));
+        }
```

Comment:
this whole thing can be replaced with a one-liner call to GSON that serializes the content, simply make sure to annotate the fields that don't match (ie: iconUrl needs to be annotated so that the name will be "icon_url" when serialized. see Hangar > Item for an example), 
You'll need either an adapter for color, or, simpler, change the field type to int and make the setter do the conversion